sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.service.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  Injectable,
+  NotFoundException,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+import { DisbursementSchedule, DisbursementValueType } from "@sims/sims-db";
+import { getTotalDisbursementAmount } from "@sims/utilities";
+import {
+  COE_NOT_FOUND_MESSAGE,
+  FIRST_COE_NOT_COMPLETE,
+  FIRST_COE_NOT_COMPLETE_MESSAGE,
+  INVALID_TUITION_REMITTANCE_AMOUNT,
+  INVALID_TUITION_REMITTANCE_AMOUNT_MESSAGE,
+} from "../../constants";
+import {
+  ApplicationService,
+  DisbursementScheduleService,
+} from "../../services";
+import { ApiProcessError } from "../../types";
+import { COE_WINDOW } from "../../utilities";
+import BaseController from "../BaseController";
+import { ConfirmEnrollmentOptions } from "./models/confirmation-of-enrollment.dto";
+
+@Injectable()
+export class ConfirmationOfEnrollmentControllerService extends BaseController {
+  constructor(
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+    private readonly applicationService: ApplicationService,
+  ) {
+    super();
+  }
+
+  /**
+   * Approve confirmation of enrollment(COE).
+   * An application can have up to two COEs based on the disbursement schedule,
+   * hence the COE approval happens twice for application with more than once disbursement.
+   * Irrespective of number of COEs to be approved, application status is set to complete
+   * on first COE approval.
+   * @param disbursementScheduleId disbursement schedule id of COE.
+   * @param auditUserId user who confirms enrollment.
+   * @param options Confirm COE options.
+   */
+  async confirmEnrollment(
+    disbursementScheduleId: number,
+    auditUserId: number,
+    options?: ConfirmEnrollmentOptions,
+  ): Promise<void> {
+    // Get the disbursement and application summary for COE.
+    const disbursementSchedule =
+      await this.disbursementScheduleService.getDisbursementAndApplicationSummary(
+        disbursementScheduleId,
+        options?.locationId,
+      );
+
+    if (!disbursementSchedule) {
+      throw new NotFoundException(COE_NOT_FOUND_MESSAGE);
+    }
+
+    if (
+      !options?.allowOutsideCOEWindow &&
+      !this.applicationService.withinValidCOEWindow(
+        new Date(disbursementSchedule.disbursementDate),
+      )
+    ) {
+      throw new UnprocessableEntityException(
+        `Confirmation of Enrollment window is greater than ${COE_WINDOW} days`,
+      );
+    }
+
+    // TODO: Add validation for past study period.
+
+    const firstOutstandingDisbursement =
+      await this.disbursementScheduleService.getFirstDisbursementSchedule({
```

Comment:
@dheepak-aot let's review the method `getFirstDisbursementSchedule`.
I am a bit confused about why we are trying to get the first pending COE but we are passing the `disbursementSchedule.id`, is it not supposed to be something like the assessment id or application id? Besides that, I guess that this method also needs to be adapted to use the application's current assessment.
Lets review it and decide if any change is needed and if we will take care of this or some other effort.